app/models/miq_server.rb

```diff
@@ -544,4 +545,16 @@ def zone_unchanged_in_pods
 
     errors.add(:zone, N_('cannot be changed when running in containers')) if zone_id_changed?
   end
+
+  def messaging_health_check
+    require 'manageiq-messaging'
+
+    _log.info("Initiating messaging health check")
+    broker = MiqQueue.messaging_client("health_check")
+    broker.publish_message(:service => "messaging-health-check", :message => "health check", :payload => {})
+    broker.subscribe_messages(:service => "messaging-health-check") { break }
```

Comment:
If we want this to work with kafka optional I think we need to handle `MiqQueue.messaging_client` being `nil` here.  Since there is nothing to check in that case you can just `return if broker.nil?`